### PR TITLE
Added section on monorepo methodology

### PR DESCRIPTION
**Pull request by Zombiefruit**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -69,3 +69,17 @@ tsconfig.base.json    // common typescript configuration
 tslint.json           // monorepo wide linting configuration
 yarn.lock             // the only lock file in the repo. all packages combined
 ```
+
+## Monorepo methodology
+
+Development in a monorepo environment should take place in the root of the project. However, users can consume specific packages (as opposed to consuming the entire monorepo) if they'd like. This methodology defines a structure for dependencies:
```
Comment (AviVahl) [inline README.md]: "consuming the entire monorepo" is a bit confusing, isn't it?

File: README.md [hunk 2]
```diff
@@ -69,3 +69,17 @@ tsconfig.base.json    // common typescript configuration
...
+- dev-dependencies should be placed in the root `package.json`
```
Comment (AviVahl) [inline README.md]: Explain why

File: README.md [hunk 3]
```diff
@@ -69,3 +69,17 @@ tsconfig.base.json    // common typescript configuration
...
+- dependencies and peer-dependencies should be placed the `package.json` of the relevant package
+
+Additionally, some packages depend on sibling packages within the monorepo. For example, `app` depends on `components`. This relationship should be described in the `package.json` of `app` like so:
+
+```json
+  "dependencies": {
+    "components": "*"
```
Comment (AviVahl) [inline README.md]: the * is only if you didn't publish yet.
technically, all packages in the monorepo got access to all packages.

File: README.md [hunk 4]
```diff
@@ -69,3 +69,17 @@ tsconfig.base.json    // common typescript configuration
 tslint.json           // monorepo wide linting configuration
 yarn.lock             // the only lock file in the repo. all packages combined
 ```
+
+## Monorepo methodology
+
+Development in a monorepo environment should take place in the root of the project. However, users can consume specific packages (as opposed to consuming the entire monorepo) if they'd like. This methodology defines a structure for dependencies:
+- dev-dependencies should be placed in the root `package.json`
+- dependencies and peer-dependencies should be placed the `package.json` of the relevant package
```
Comment (AviVahl) [inline README.md]: Explain why

File: README.md (hunk 1 above)
Comment (Zombiefruit) [reply README.md]: "as opposed to adding the entire monorepo as a dependency"?

File: README.md (hunk 3 above)
Comment (Zombiefruit) [reply README.md]: So add that as a note?

Comment (AviVahl) [reply README.md]: It's just not clarified enough... the * is temporary until ypu publish. Then, it becomes an actual compatible version (when publishing). this probably needs to be explained.

Comment (AviVahl) [reply README.md]: @Zombiefruit removed the * from master.

Now they specify versions even if not published. It just makes sense.

Comment (Zombiefruit) [reply README.md]: Sweet. I'll update ui-autotools to do the same

**Comment by AviVahl** [comment]
@Zombiefruit please make sure to squash changes into a single commit next time, rather than merging the commits as-is to master.

**Pull request merged**